src/utils/file/file.util.ts

```
@@ -2,201 +2,221 @@ import * as fs from 'fs';
+        // Remove page numbers and headers/footers (basic patterns)
+        .replace(/^\d+\s*$/gm, '')
+        // Remove isolated single characters that might be artifacts
+        .replace(/\n\s*[a-zA-Z]\s*\n/g, '\n')
+        // Fix hyphenated words split across lines
+        .replace(/(\w+)-\s*\n\s*(\w+)/g, '$1$2')
+        // Normalize line breaks
+        .replace(/\n{3,}/g, '\n\n')
+        // Remove common PDF artifacts
+        .replace(/\f/g, '\n') // Form feed to line break
+        // eslint-disable-next-line no-control-regex
+        .replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, '') // Remove control characters
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)